Update 1.5.0

starstormhun opened the pull request:
- Included Unity Assets for easier replication
- Added DisablePointLights slider to all materials
- Added HairPlusReflect
- Added HairFrontPlusReflect

starstormhun committed:
Update 1.4.0

- Added DisablePointLights slider to all materials
- Added HairPlusReflect
- Added HairFrontPlusReflect

starstormhun committed:
Update .gitignore

starstormhun committed:
Ignore meta

starstormhun committed:
Update 1.4.1

- Added a slider to disable matcaps which fall in shadow
- Fixed MainOpaquePlusTess not working

markus-tamas commented:
Updated to 1.4.1

starstormhun commented:
- Added slider to disable matcaps which fall in shadow
- Fixed MainOpaquePlusTess not working

starstormhun committed:
v1.4.2

Added:
- All shaders now support matcap shadowing
- Reflection coloring with 2 options sliders (ReflectCol, ReflectColAlphaOpt, ReflectColColorOpt)
- ReflectRotation slider: 0-360° matcap rotation
- ReflectMask for Skin/Hair reflect versions: black/white toggles between environmental and matcap reflections, defaults to white (matcap)
- ReflectMapDetail for item shaders: combination of ReflectMap (red channel) and ReflectMask (green channel)
 - Had to combine them due to technical limitations
- ShadowHSV: shadow color relative adjustment (for touchups)
- AdjustBackfaceNormals slider: Backfaces will be lit correctly but the item's reflection looks off (defaults to off, on would be the vanilla behavior)
- CullOption and Cutoff sliders for hair shaders
- All item shaders which didn't before now support AlphaMasks
- Alpha shaders now support smooth alpha (AlphaOptionCutoff = 2)
 - Gray AlphaMask values between `Cutoff` and 1 will be scaled to between 0 and 1
Fixes:
- Point lights now work correctly with matcap shadowing
- ReflectVal inconsistencies compared to 1.3.1 removed
- Matcap alphas now behave like in 1.3.1
- Front hair shaders now have the correct render queue (might mean needing to set "Show irises through hair" to Yes)

starstormhun commented:
Added:
- All shaders now support matcap shadowing
- Reflection coloring with 2 options sliders (ReflectCol, ReflectColAlphaOpt, ReflectColColorOpt)
- ReflectRotation slider: 0-360° matcap rotation
- ReflectMask for Skin/Hair reflect versions: black/white toggles between environmental and matcap reflections, defaults to white (matcap)
- ReflectMapDetail for item shaders: combination of ReflectMap (red channel) and ReflectMask (green channel)
 - Had to combine them due to technical limitations
- ShadowHSV: shadow color relative adjustment (for touchups)
- AdjustBackfaceNormals slider: Backfaces will be lit correctly but the item's reflection looks off (defaults to off, on would be the vanilla behavior)
- CullOption and Cutoff sliders for hair shaders
- All item shaders which didn't before now support AlphaMasks
- Alpha shaders now support smooth alpha (AlphaOptionCutoff = 2)
 - Gray AlphaMask values between `Cutoff` and 1 will be scaled to between 0 and 1
Fixes:
- Point lights now work correctly with matcap shadowing
- ReflectVal inconsistencies compared to 1.3.1 removed
- Matcap alphas now behave like in 1.3.1
- Front hair shaders now have the correct render queue (might mean needing to set "Show irises through hair" to Yes)

starstormhun committed:
Update .gitignore

starstormhun committed:
v1.4.3

starstormhun commented:
Added:
- ReflectiveOverlayed is now back in, as xukmi's source did not contain it for whatever reason
- Matcap scaling and offset now works
Fixed:
- shadowcolor's alpha being set to 0 was causing trouble
- adding alphamasks to things that weren't supposed to have them was a grave mistake and has been reverted, bras now work again
- sometimes pure black colors were causing problems, so I made sure never to return pure black

starstormhun committed:
whitespace adjustment

starstormhun committed:
Sync with unity project

starstormhun committed:
Add eye matcaps, fix eye Z rendering

starstormhun committed:
Merge pull request #1 from starstormhun/AddEyeReflections

Add eye matcaps, fix eye Z rendering

starstormhun committed:
Add eye tess shaders, remove shaderforgeexample

starstormhun committed:
Switch eye shaders to separate sampler base

starstormhun committed:
Change Skin shaders to use separate samplers

starstormhun committed:
Change Item shaders to use separate sampler

starstormhun committed:
Convert Unity sampler defines to homebrew defines

starstormhun committed:
Add tess versions of hair shaders

starstormhun committed:
Add tess versions of hair and missing item tess shaders

starstormhun committed:
Separate item shaders' fragment shader to file

starstormhun committed:
Fix eye matcap eye tracking

starstormhun committed:
Merge pull request #2 from starstormhun/AddMoreTess

Add more tess

starstormhun committed:
remove specularmap from manifest

starstormhun committed:
meta update

starstormhun committed:
Update prefabs to fix eye layering

starstormhun committed:
Setup basic shader (copy of mainitem) and inputs

starstormhun committed:
Make material and prefab

starstormhun committed:
Implement patterns and offset/rotation

starstormhun committed:
Implement pattern clamping

starstormhun committed:
Fix shading and normals

starstormhun committed:
Add alpha shader

starstormhun committed:
ShadowExtend trials

starstormhun committed:
Add transparency slider to front hairs

starstormhun committed:
Add meta files

starstormhun committed:
Implement ambientshadowOFF, fix detailmask.g

starstormhun committed:
Fix linemask

starstormhun committed:
Merge pull request #3 from starstormhun/Studio-Item-Plus

Add Studio item plus

starstormhun committed:
Add tess variants of studio shaders

starstormhun committed:
Recompile shaders for 1.5.0

starstormhun committed:
Fix reported 1.5.0 bugs

- fix eye matcaps extending beyond maintex
- fix absolute black problem with eye / eyetess
- fix transparency slider default
- fix transparency slider behaviour with outlines / reflections
- fix absolute black shadow color problem with studio shaders

starstormhun commented:
- fixed skin/item matcap behaviour, now completely backwards compatible with 1.3.1
- removed ReflectColAlphaOpt and ReflectColColorOpt because they were deprecated, now replaced with ReflectColMix (defaults to 1, full multiplication)
- fixed KKPrim having a dark rim where it was supposed to fade into the underlying color
- fixed AdjustBackFaceNormals not working on some shaders
- added rimReflectMode slider (controls where rim lighting happens: -1 only in shadows, 0 default, 1 only in light)
- matcap support for eye shaders
- fixed issue with eye shaders rendering in incorrect order
- added Tess versions of: EyePlus, EyeWPlus, HairPlus, HairPlusReflect, HairFrontPlus, HairFrontPlusReflect, MainItemPlus, MainItemAlphaPlus
- fixed issue with eye shaders rendering in incorrect order, again
- added transparency slider to front hairs (so you can see eyes through them, need to enable "Show eyes through hair" for this to have any effect)
- implemented MainItemStudioPlus, MainItemStudioAlphaPlus and their tess equivalents
- fixed even more issues with eye shaders
- Cutoff now works with EyeWPlus and EyeWPlusTess

starstormhun committed:
Shader recompile for release

starstormhun committed:
Update KKPItemFrag.cginc

starstormhun committed:
Fix Item reflections and LineTex tiling on some clothing

starstormhun committed:
Add specular normal scaling

starstormhun committed:
Update manifest.xml

starstormhun committed:
Add back culling to transparent front hair parts

starstormhun committed:
fix outline hole problem with transparent hairs

starstormhun committed:
Fix cum overlay tiling

starstormhun committed:
Expose AlphaMask on skin shaders

starstormhun committed:
v1.5.2

## Additions:
- Exposed AlphaMask on Skin shaders
- Added ColMask / Col0 / Col1 / Col2 / Col3 to Skin shaders
## Fixes:
- Fixed all tiling issues ||hopefully||
- Eye emission now rotates with the eye
- Eye emission is no longer broken by too dark of a MainTex
- Fix hair reflections breaking things when parts of hair were hidden via MainTex
- Skin / Hair / Eye shaders' ReflectMap now functions correctly
- Skin / Hair / Eye shaders' ReflectMask can now be scaled/translated
- A bunch of StudioItem shader color nonsense
- StudioItem shaders now only speculate in the light
- StudioItem shaders are now properly lit with Point Lights (pointlight shadows still impossible)
- StudioItem shaders' speculars fixed all around
- StudioItem shaders' emission fixed

starstormhun committed:
Fix reported problems, recompile for release

starstormhun committed:
Fix version

starstormhun committed:
Reorganise Sampler/Texture2D stuff

starstormhun committed:
Increment version to 1.5.3

RikkiBalboa committed:
Fix casted shadows not having an effect on DisableShadowedMatcap

starstormhun committed:
Merge pull request #4 from RikkiBalboa/fix-disable-matcap-in-shadow

Fix DisableShadowedMatcap with cast shadows

starstormhun committed:
Bump version

bigqingshu commented:
If possible, please add the ‘AlphaOptionCutoff’ feature to the SkinPlusTess shader. This feature would be helpful for me. Thank you!

RikkiBalboa committed:
Add emission mask mode

starstormhun committed:
Merge pull request #5 from RikkiBalboa/add-emission-mask-mode

Add different emission mask mode

starstormhun committed:
Bump version

starstormhun committed:
Add Saturation slider to all shaders

Strawberry-kun committed:
ColorMask Materials

Strawberry-kun committed:
ColorMask Shaders

Strawberry-kun committed:
ColorMask Prefabs

Strawberry-kun committed:
ColorMask Manifest Update

Strawberry-kun committed:
Added saturation to shaders

Strawberry-kun committed:
Update manifest

Strawberry-kun committed:
Merge branch 'main' into colormasks

Strawberry-kun committed:
Indenting

starstormhun committed:
Merge pull request #6 from Strawberry-kun/colormasks

Add Alpha/Opaque shaders with Color Mask support

starstormhun committed:
Fix eye emissions and bump version

gievepix committed:
ColMask support for clothing shaders

...instead of adding separate shaders

starstormhun committed:
Merge pull request #7 from gievepix/colmask

ColMask support for clothing shaders instead of extra shaders

starstormhun committed:
Fix version, try adding Clock default value in manifest

starstormhun committed:
v1.7.2

- Add EmissionKeepCol slider to all shaders that toggles between 1.5.4 and 1.7.1 emission behaviour
- this slider defaults to 1 on eye shaders and 0 on everything else

starstormhun committed:
Update README.md
